newsfragments/41814.significant.rst

```diff
@@ -0,0 +1 @@
+Change the default value of ``strict_dataset_uri_validation`` to True.
```

Comment:
We should mention somewhere that non-URI-based datasets should use `name` instead, but that hasn’t been implemented so I’m not sure if it’s a good idea to mention it here. But if we don’t, we might fotget and not mentioning this at all in the end.

Reply:
I feel like it does not really make sense to mention here 🤔 this one is not actually blocking anything. or maybe we could wait for that to be implemented and then merge this one (with `using name` mentioned)

Reply:
Discussed with Wei earlier. This error only matters when you use certain URI schemes that have specific semantics (e.g. Postgres, S3).  fter AIP-72, you still need to provide a URI when the asset is named _when this error message matters_ (the entire point if for you to provide a meaningful URI. So the better way to help users migrate would be to

1. Make this an error (already done in this PR)
2. Add documentation and add a link in the error message to point users to the right directions.
    * If you can, provide a good URI for the resource. The `name` argument can optionally be used to provide a more human-readable identifier to the asset.
    * If you can’t provide a meaningful URI, use another scheme altogether that don’t have the semantic restrictions. Airflow will never require a semantic for user-defined URI schemes (with a prefix `x-`), so that can be a good alternative.
    * If you don’t have a URI, consider using `AssetAlias` instead.

Reply:
I just updated it along with some documentation fix